client/src/models/Report.js

```diff
@@ -410,4 +410,62 @@ export default class Report extends Model {
     })
     return { taskToAssessmentUuid, taskAssessments }
   }
+
+  hasConflict(that) {
```

Comment:
seems like most of what this method do is finding if 2 time windows overlap. Should be moved somewhere else as a time utility

Reply:
I am afraid I don't agree. What most of this method do is determining start and end dates of two reports based on duration/all-day status. Only last statement finds if 2 time windows overlap. If we convert it to a time utility we might need to repeat start/end date related logic wherever we need this functionality. I insist on keeping it as it is.

However, there is no point of making it an instance method, it can be converted to a static function. Such that;

```js
  static hasConflict(report01, report02) {
    if (report01.uuid === report02.uuid) {
      return false // same report is not a conflicting report
    }

    let start01
    let end01

    if (report01.duration) {
      start01 = moment(report01.engagementDate)
      end01 = moment(report01.engagementDate).add(report01.duration, "minute")
    } else {
      // It is an all-day event if duration is not given
      start01 = moment(report01.engagementDate).startOf("day")
      end01 = moment(report01.engagementDate).endOf("day")
    }

    let start02
    let end02

    if (report02.duration) {
      start02 = moment(report02.engagementDate)
      end02 = moment(report02.engagementDate).add(report02.duration, "minute")
    } else {
      // It is an all-day event if duration is not given
      start02 = moment(report02.engagementDate).startOf("day")
      end02 = moment(report02.engagementDate).endOf("day")
    }

    return (
      start01.isSame(start02) ||
      (end01.isAfter(start02) && start01.isBefore(end02))
    )
  }
```